### DISCUSSION TITLE
SMTP email not error

### DISCUSSION THREAD

**Discussion by dieriba** [directus/directus#15967] [Archive]
### Describe the Bug

When I configure my .env file to send email with flows, I have an error once the flow is triggered.
When I use hooks with nodemailer I have the same error unless I add the options [tls : {rejected}](tls: { rejectUnauthorized: false })
But in the .env we can't setup this options.

here the working code with nodemailer : 

```js
const nodemailer = require('nodemailer');

module.exports = function registerHook({ action }) {
  action('items.create', async () => {
   let transporter = nodemailer.createTransport({
    host: "myhost",
    port: 587,
    secure: false, // true for 465, false for other ports
    auth: {
      user: 'email@email.com', // generated ethereal user
      pass: 'mypassword"// generated ethereal password
    },
  });

  // send mail with defined transport object
  try {
    await transporter.sendMail({
    from: "email@email.com", // sender address
    to: "t.dieriba@gmail.com", // list of receivers
    subject: "Hello ✔", // Subject line
    text: "Hello world?", // plain text body
    html: "<b>Hello world?</b>", // html body
  });
  } catch (error) {
    console.log(error);
  }
    return;
  });
};
```

### To Reproduce

to reproduce the bug, you either choose one of the port 465 or 587.

### Errors Shown

If I put the port 587 and EMAIL_SMTP_SECURE=false I get this error :

```json
 {
  "library": "SSL routines",
  "function": "ssl3_get_record",
  "reason": "wrong version number",
  "code": "ESOCKET",
  "command": "CONN"
}
```

if I put the port 465 and EMAIL_SMTP_SECURE=true i get this error: 

```json
{
  "code": "EENVELOPE",
  "command": "API"
}
```

### What version of Directus are you using?

9.18.1

### What version of Node.js are you using?

16.18.0

### What database are you using?

SQLite 3.31.0

### What browser are you using?

Edge

### How are you deploying Directus?

locally

**Comment by rijkvanzanten** [comment]
You should be able to setup the same config you're using in your code example in the env vars:

```
EMAIL_TRANSPORT="smtp"
EMAIL_SMTP_HOST="smtp.ataera.world"
EMAIL_SMTP_PORT="587"
EMAIL_SMTP_SECURE="false"
EMAIL_SMTP_USER="no-reply@ataera.world"
EMAIL_SMTP_PASSWORD="kuthu5-wubkyx-V"
```

**Reply by dieriba** [reply]
Here what I get again : 
{
  "library": "SSL routines",
  "function": "ssl3_get_record",
  "reason": "wrong version number",
  "code": "ESOCKET",
  "command": "CONN"
}
would you mind edit your message to add fake value inside the env pls

**Reply by dieriba** [reply]
In nodemailer i had to add tls : {rejectUnauthirizes : false}
How can i make the same with dir ?

**Reply by connorwinston** [reply]
To add extra environment variables that are nested, check out this are in the docs
https://docs.directus.io/self-hosted/config-options.html#type-casting-and-nesting

So since `tls` needs to be passed into nodemailer, try adding the env variable `EMAIL_SMTP_TLS__REJECT_UNAUTHORIZED` to change that setting.

You can find the smtp TLS being passed into nodemailer here, https://github.com/directus/directus/blob/31a6fd208ffcaedf128d98be8940f172b635b276/api/src/mailer.ts#L39

**Reply by notflip** [reply]
@dieriba Where can you see those logs?
